react/components/ComponentEditor.js

```diff
@@ -74,9 +72,21 @@ class ComponentEditor extends Component {
       return null
     }
 
+    const getDeepProps = (properties = {}, prevProps) =>
```

Comment:
Add JSDoc explaining really well whats going on that logic and why you needed to do that. Follow the example of the function `getUiSchema`